app/views/layouts/angular-bootstrap/_endpoints_angular.html.haml

```diff
@@ -223,14 +224,32 @@
       %textarea.form-control{"id"          => "#{prefix}_tls_ca_certs",
                              "name"        => "#{prefix}_tls_ca_certs",
                              "ng-model"    => "#{ng_model}.#{prefix}_tls_ca_certs",
-                             "ng-disabled" => "emsCommonModel.emstype == 'rhevm' && !#{ng_model}.#{prefix}_tls_verify",
+                             "ng-disabled" => "(emsCommonModel.emstype == 'rhevm' || emsCommonModel.emstype == 'kubevirt') && !#{ng_model}.#{prefix}_tls_verify",
                              "ng-required" => false,
                              "ng-trim"     => false,
                              "prefix"      => prefix.to_s,
                              "reset-validation-status" => "#{prefix}_auth_status"}
       %span.help-block
         = _("Paste here the trusted CA certificates, in PEM format.")
 
+%div{"ng-if"=> "emsCommonModel.emstype == 'kubevirt'"}
+  .form-group{"ng-class" => "{'has-error': angularForm.#{prefix}_kubevirt_token.$invalid}"}
+    %label.col-md-2.control-label{"for" => "#{prefix}_kubevirt_token"}
+      = _('Token')
+    .col-md-4
+      %input.form-control{"type"                    => "text",
+                          "id"                      => "#{prefix}_kubevirt_token",
```

Comment:
Again, we should not be doing provider specifics here.  There are already ways to have token-based authentication and we should leverage that instead of creating a new section, I would think.

Reply:
You can remove this div completely.

The provider specifics should arrive from the haml that is one level above this - multi_auth_credentials.html.haml.

Reply:
Refer to this div `%div{"ng-if" => controller_name == "ems_datawarehouse"}` in multi_auth_credentials.html.haml to get an idea of how to pass the provider specific parameters to the endpoints haml and credentials haml

P.S. Let me know if you need the exact changes to be done in multi_auth_credentials.html.haml - I can send you a patch.